.github/workflows/docker_build.yml

```
@@ -1,18 +1,59 @@
-name: Publish Docker
-on: 
+# -------------------------------------------------------------------------------------------------
+# Job Name
+# -------------------------------------------------------------------------------------------------
+name: Build-Publish
+
+# -------------------------------------------------------------------------------------------------
+# When to run
+# -------------------------------------------------------------------------------------------------
+on:
   push: {}
   schedule:
     - cron:  '0 0 * * *'
+
+# -------------------------------------------------------------------------------------------------
+# What to run
+# -------------------------------------------------------------------------------------------------
 jobs:
   build-publish:
+    name:  "[ ${{ matrix.TF_VERSION }}-${{ matrix.TG_VERSION }} ]"
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: False
+      matrix:
+        TF_VERSION: [0.12,0.13,latest]
+        TG_VERSION: [0.21.7,0.22,latest]
```

I'd also make this more dynamic via:

```suggestion
        TG_VERSION: [0.21,0.22,0.23,latest]
```

---

We are still tied to version `0.21.7` if I remove this version, we will jump to the latest patch level of`0.21`